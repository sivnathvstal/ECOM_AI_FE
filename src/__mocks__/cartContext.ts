import { Cart, CartProduct } from '../types';

export const mockCartItems: CartProduct[] = [
  {
    productId: 1,
    quantity: 2,
    product: {
      id: 1,
      title: 'Test Product 1',
      price: 10.99,
      description: 'Test description 1',
      category: 'electronics',
      image: 'test1.jpg',
      rating: { rate: 4.5, count: 100 },
    },
  },
  {
    productId: 2,
    quantity: 1,
    product: {
      id: 2,
      title: 'Test Product 2',
      price: 20.99,
      description: 'Test description 2',
      category: 'clothing',
      image: 'test2.jpg',
      rating: { rate: 4.0, count: 80 },
    },
  },
];

export const mockCart: Cart = {
  id: 1,
  userId: 1,
  date: '2023-01-01',
  products: [
    { productId: 1, quantity: 2 },
    { productId: 2, quantity: 1 },
  ],
};

export const mockCartContext = {
  cart: mockCart,
  cartItems: mockCartItems,
  isLoading: false,
  error: null,
  addItem: jest.fn(),
  updateItem: jest.fn(),
  removeItem: jest.fn(),
  totalItems: 3, // 2 + 1
  totalPrice: 42.97, // 10.99*2 + 20.99*1
};

export const mockEmptyCartContext = {
  cart: {
    id: 0,
    userId: 1,
    date: '2023-01-01',
    products: [],
  },
  cartItems: [],
  isLoading: false,
  error: null,
  addItem: jest.fn(),
  updateItem: jest.fn(),
  removeItem: jest.fn(),
  totalItems: 0,
  totalPrice: 0,
};

export const mockLoadingCartContext = {
  ...mockCartContext,
  isLoading: true,
};

export const mockErrorCartContext = {
  ...mockCartContext,
  error: 'Failed to fetch cart',
};

const defaultExport = {
  mockCartItems,
  mockCart,
  mockCartContext,
  mockEmptyCartContext,
  mockLoadingCartContext,
  mockErrorCartContext,
};
export default defaultExport; 