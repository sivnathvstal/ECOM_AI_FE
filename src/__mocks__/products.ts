import { Product, PaginationParams, ProductsResponse } from '../types';

export const mockProducts: Product[] = [
  {
    id: 1,
    title: 'Test Product 1',
    price: 10.99,
    description: 'Test description 1',
    category: 'electronics',
    image: 'test1.jpg',
    rating: { rate: 4.5, count: 100 },
  },
  {
    id: 2,
    title: 'Test Product 2',
    price: 20.99,
    description: 'Test description 2',
    category: 'clothing',
    image: 'test2.jpg',
    rating: { rate: 4.0, count: 80 },
  },
  {
    id: 3,
    title: 'Test Product 3',
    price: 15.99,
    description: 'Test description 3',
    category: 'electronics',
    image: 'test3.jpg',
    rating: { rate: 3.5, count: 60 },
  },
  {
    id: 4,
    title: 'Test Product 4',
    price: 25.99,
    description: 'Test description 4',
    category: 'jewelry',
    image: 'test4.jpg',
    rating: { rate: 4.8, count: 120 },
  },
];

export const mockPagination: PaginationParams = {
  page: 1,
  limit: 8,
  total: 4,
};

export const mockProductsResponse: ProductsResponse = {
  products: mockProducts,
  pagination: mockPagination,
};

export const mockCategories: string[] = [
  'electronics',
  'clothing',
  'jewelry',
]; 