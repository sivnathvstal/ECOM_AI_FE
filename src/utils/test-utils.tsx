import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { ThemeProvider, createTheme } from '@material-ui/core';
import { QueryClient, QueryClientProvider } from 'react-query';
import { CartProvider } from '../context/CartContext';

// Create a theme for testing
const theme = createTheme({
  palette: {
    primary: {
      main: '#3f51b5',
    },
    secondary: {
      main: '#f50057',
    },
  },
});

// Create a client for React Query
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

// All providers wrapper
const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={theme}>
        <CartProvider>
          <BrowserRouter>{children}</BrowserRouter>
        </CartProvider>
      </ThemeProvider>
    </QueryClientProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// Re-export everything
export * from '@testing-library/react';

// Override render method
export { customRender as render }; 