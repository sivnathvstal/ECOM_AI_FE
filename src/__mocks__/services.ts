import { mockProducts, mockProductsResponse, mockCategories } from './products';
import { mockCart, mockCartItems } from './cartContext';

// Product service mocks
export const mockProductService = {
  getProducts: jest.fn().mockResolvedValue(mockProductsResponse),
  getProductById: jest.fn().mockImplementation((id: number) => {
    const product = mockProducts.find(p => p.id === id);
    if (product) {
      return Promise.resolve(product);
    }
    return Promise.reject(new Error('Product not found'));
  }),
  getCategories: jest.fn().mockResolvedValue(mockCategories),
  searchProducts: jest.fn().mockResolvedValue(mockProductsResponse),
};

// Cart service mocks
export const mockCartService = {
  getCart: jest.fn().mockResolvedValue(mockCart),
  addToCart: jest.fn().mockResolvedValue(mockCart),
  updateCartItem: jest.fn().mockResolvedValue(mockCart),
  removeFromCart: jest.fn().mockResolvedValue(mockCart),
  getCartWithProducts: jest.fn().mockResolvedValue(mockCartItems),
}; 