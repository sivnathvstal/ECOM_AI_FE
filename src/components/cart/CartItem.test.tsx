import React from 'react';
import { render, screen, fireEvent } from '../../utils/test-utils';
import CartItemComponent from './CartItem';
import { mockCartItems } from '../../__mocks__/cartContext';

describe('CartItemComponent', () => {
  const mockOnRemove = jest.fn();
  const cartItem = mockCartItems[0];
  
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { container } = render(
      <CartItemComponent 
        item={cartItem} 
        onRemove={mockOnRemove} 
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('displays item information correctly', () => {
    render(
      <CartItemComponent 
        item={cartItem} 
        onRemove={mockOnRemove} 
      />
    );
    
    const { productId, quantity, product } = cartItem;
    
    expect(screen.getByTestId(`cart-item-title-${productId}`)).toHaveTextContent(product.title);
    expect(screen.getByTestId(`cart-item-quantity-${productId}`)).toHaveTextContent(`Quantity: ${quantity}`);
    expect(screen.getByTestId(`cart-item-price-${productId}`)).toHaveTextContent(`$${(product.price * quantity).toFixed(2)}`);
    
    const image = screen.getByTestId(`cart-item-image-${productId}`);
    expect(image).toHaveAttribute('src', product.image);
    expect(image).toHaveAttribute('alt', product.title);
  });

  it('calls onRemove when remove button is clicked', () => {
    render(
      <CartItemComponent 
        item={cartItem} 
        onRemove={mockOnRemove} 
      />
    );
    
    const removeButton = screen.getByTestId(`cart-item-remove-${cartItem.productId}`);
    fireEvent.click(removeButton);
    
    expect(mockOnRemove).toHaveBeenCalledWith(cartItem.productId);
  });

  it('returns null when product is undefined', () => {
    const itemWithoutProduct = { ...cartItem, product: undefined };
    const { container } = render(
      <CartItemComponent 
        item={itemWithoutProduct} 
        onRemove={mockOnRemove} 
      />
    );
    
    expect(container.firstChild).toBeNull();
  });
}); 