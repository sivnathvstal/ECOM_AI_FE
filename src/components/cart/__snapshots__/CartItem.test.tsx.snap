// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItemComponent renders correctly 1`] = `
<div>
  <div
    class="MuiPaper-root cart-item MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="cart-item-1"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <img
          alt="Test Product 1"
          class="cart-item-image"
          data-testid="cart-item-image-1"
          src="test1.jpg"
        />
      </div>
      <div
        class="MuiGrid-root cart-item-info MuiGrid-item MuiGrid-grid-xs-6"
      >
        <h6
          class="MuiTypography-root cart-item-title MuiTypography-subtitle1"
          data-testid="cart-item-title-1"
        >
          Test Product 1
        </h6>
        <div
          class="MuiBox-root MuiBox-root-1 cart-item-quantity"
          data-testid="cart-item-quantity-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Quantity: 
            2
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root cart-item-actions MuiGrid-item MuiGrid-grid-xs-3"
      >
        <p
          class="MuiTypography-root cart-item-price MuiTypography-body1"
          data-testid="cart-item-price-1"
        >
          $
          21.98
        </p>
      </div>
      <div
        class="MuiGrid-root cart-item-actions MuiGrid-item MuiGrid-grid-xs-1"
      >
        <button
          aria-label="delete"
          class="MuiButtonBase-root MuiIconButton-root"
          data-testid="cart-item-remove-1"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
